lib/git_clerk.rb

```diff
@@ -1,17 +1,43 @@
-require_relative 'git_clerk/dir_processor'
-require_relative 'git_clerk/visualizer'
+require 'git_clerk/dir_processor'
+require 'git_clerk/visualizer'
+require 'git_clerk/opt_parser'
+require 'git_clerk/errors/main_command_not_specified_error'
+require 'git_clerk/errors/options_not_compatible_error'
+require 'git_clerk/errors/option_instead_of_argument_error'
+require 'pry'
 
 module GitClerk
   include Gitable
-  def self.clerk
+
+  DEFAULT_OPTIONS = {
+    verbose: false,
+    dir: Dir.pwd,
+    show_dirty: false
+  }.freeze
+
+  def self.clerk_and_print
+    parser = OptParser.new
+    parser.parse!
+    options = parser.result
+    pp options
+  rescue StandardError => e
+    puts e.message
+
+    # ---- TODO: decomment it back after pry tests on options!!!
+    # processor = DirProcessor.new(Dir.pwd)
+    # processor.clerk
+    # visualize(processor.data)
+    # ---- ENDS HERE ----
+
+    # result = clerk(options)
+    # visualize(result.data)
+  end
+
+  def self.clerk(options = DEFAULT_OPTIONS)
     # TODO: take dir from argv || use Dir.pwd
-    dir = '/Users/nstoianov/dev/'
-    processor = DirProcessor.new(dir)
+    # processor = DirProcessor.new(options)
```

Comment:
`DirProcessor` to handle new hash and its values
Also consider a class/instance before `DirProcessor` to verify the main command and call relevant code

Reply:
Done, `CommandRunner` is implemented for handling commands